assets/js/updaters.js

```diff
@@ -886,6 +887,8 @@ function updateInvasions() {
     }
 
     invasions.forEach(invasion => {
+      // store current invasion ids
+      invasionIDs.push(invasion.id);
```

Comment:
Given your findings on SO, I would definitely recommend switching to an object since its performance is better, at no cost of code complexity.